op-challenger/fault/player.go

```diff
@@ -94,6 +94,15 @@ func NewGamePlayer(
 }
 
 func (g *GamePlayer) ProgressGame(ctx context.Context) bool {
+	status, err := g.caller.GetGameStatus(ctx)
+	if err != nil {
+		g.logger.Warn("Unable to retrieve game status", "err", err)
+		return false
+	}
```

Comment:
We are essentially doing this twice in this function (before and after `Act`ing) so maybe we can refactor it out into a `getGameStatus` method on the `GamePlayer` that can dedupe the logging at least.

Reply:
I'm not entirely sure what you mean here - in my mind this _is_ the code that performs the status check before acting and exits early.

Reply:
Discussed this separately. There's a couple of factors here:
1. We would ideally like to avoid extra RPC calls
2. We want to log the final game result once and then stop logging anything about the game

So going to change this to cache the completed status rather than requesting the status before and after acting.